task02/main.cpp

```diff
@@ -1,27 +1,63 @@
 
 #include <iostream>
+#include "queue.cpp"
```

Comment:
It is the wrong pattern, never use it! 
We include header files and compile code (.cpp) files. So put your templates implementation into the header (you could read about template instantiation and code generation to understand details).